File: sample-projects/vitest-project/string.test.js
import { describe, it, expect } from 'vitest';

describe('String operations', () => {
  it('should concatenate strings', () => {
    console.log('Testing string concatenation...');
    const result = 'Hello' + ' ' + 'World';
    expect(result).toBe('Hello World');
    console.log('String concatenation passed!');
  });

  it('should fail this test', () => {
    console.log('This test is expected to fail');
    console.error('Error: Intentional failure for testing');
    expect('foo').toBe('bar'); // This will fail
  });

  it.skip('should skip this test', () => {
    console.log('This should not run');
    expect(true).toBe(false);
  });

  it('should convert to uppercase', () => {
    console.log('Testing uppercase conversion...');
    expect('hello'.toUpperCase()).toBe('HELLO');
    console.log('Uppercase test passed!');
  });
});

File: tests/fixtures/basic-go/string_test.go
package main

import (
	"strings"
	"testing"
)

func TestStringOperations(t *testing.T) {
	t.Run("concatenation", func(t *testing.T) {
		result := strings.Join([]string{"hello", "world"}, " ")
		if result != "hello world" {
			t.Errorf("got %q, want %q", result, "hello world")
		}
	})

	t.Run("uppercase", func(t *testing.T) {
		result := strings.ToUpper("hello")
		if result != "HELLO" {
			t.Errorf("got %q, want %q", result, "HELLO")
		}
	})

	t.Run("contains", func(t *testing.T) {
		if !strings.Contains("hello world", "world") {
			t.Error("expected string to contain 'world'")
		}
	})
}

func TestParallelTests(t *testing.T) {
	t.Run("parallel test 1", func(t *testing.T) {
		t.Parallel()
		// Simulate some work
		result := len("test")
		if result != 4 {
			t.Errorf("got %d, want 4", result)
		}
	})

	t.Run("parallel test 2", func(t *testing.T) {
		t.Parallel()
		// Simulate some work
		result := strings.Count("hello", "l")
		if result != 2 {
			t.Errorf("got %d, want 2", result)
		}
	})

	t.Run("parallel test 3", func(t *testing.T) {
		t.Parallel()
		// Simulate some work
		result := strings.Index("world", "r")
		if result != 2 {
			t.Errorf("got %d, want 2", result)
		}
	})
}
